### ISSUE TITLE
Label stratification in batches

### ISSUE THREAD

**Issue by paperplaneuser** [vlfeat/matconvnet#393]
Hi,
This is not an issue but a question. I'm sorry for this but I can't think of a better place to ask this.

I have a dataset that is ordered with labels, so it goes like [1,1,1, ... 2,2,2, .... , C,C,C] etc. so in this case usually all batches have examples from one class. so one batch is of class 1, the other is class 2, etc. Is this OK or do I need to shuffle the data so that each batch has a similar distribution of classes? 

Thanks for your time,
Sincerely

**Comment by nicjac** [comment]
If you are using the built-in training functions (e.g. "cnn_train_dag.m" or "cnn_train.m"), the labels are automatically shuffled before training. Having batches that consist of a single class would be detrimental to the learning process.

**Comment by paperplaneuser** [comment]
yes i'm using the cnn_train function, and I implemented my own getBatch function, but i never checked the batch variable (i.e. the image indices) that is supplied to it. so if it's already shuffled, then i don't need to worry :)
thanks for your answer!
